core/store/migrate/migrate.go

```diff
@@ -0,0 +1,82 @@
+package migrate
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	_ "github.com/smartcontractkit/chainlink/core/store/migrate/migrations" // Invoke init() functions within migrations pkg.
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/pressly/goose/v3"
+)
+
+//go:embed migrations/*.sql
+var embedMigrations embed.FS
+
+const MIGRATIONS_DIR string = "migrations"
+
+func init() {
+	goose.SetBaseFS(embedMigrations)
+	goose.SetSequential(true)
+	goose.SetTableName("goose_migrations")
+	// goose.SetVerbose(true) // can be set to debug migrations
+}
+
+// Ensure we migrated from v1 migrations to goose_migrations
+func ensureMigrated(db *sql.DB) {
+	var count int
+	err := db.QueryRow(`SELECT count(*) FROM migrations`).Scan(&count)
+	if err != nil {
+		// already migrated
+		return
+	}
+
+	// ensure a goose migrations table exists with it's initial v0
+	if _, err = goose.GetDBVersion(db); err != nil {
+		panic(err)
+	}
+
+	// insert records for existing migrations
+	sql := `INSERT INTO %s (version_id, is_applied) VALUES %s;`
+	valueStrings := []string{}
+	for i := 1; i <= count; i++ {
+		valueStrings = append(valueStrings, fmt.Sprintf("(%v, true)", strconv.FormatInt(int64(i), 10)))
+	}
+	sql = fmt.Sprintf(sql, goose.TableName(), strings.Join(valueStrings, ","))
+
+	err = postgres.SqlTransaction(context.Background(), db, func(tx *sqlx.Tx) error {
+		if _, err = db.Exec(sql); err != nil {
+			return err
+		}
+
+		_, err = db.Exec("DROP TABLE migrations;")
+		return err
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func Migrate(db *sql.DB) error {
```

Comment:
I assume you have tested on at least one aged production database?

Reply:
I tested it on my local db with:

- An empty db (`reset`)
- A fully migrated db that used the old migrations (I ran reset on the develop branch, switched over then ran migrate)
- A partially migrated db that used the old migrations (I ran reset after deleting a couple migrations on develop, then switched over)